core/src/main/java/org/elasticsearch/cluster/action/shard/ShardStateAction.java

```diff
@@ -334,10 +377,7 @@ public String toString() {
         default void onSuccess() {
         }
 
-        default void onShardFailedNoMaster() {
-        }
-
-        default void onShardFailedFailure(final DiscoveryNode master, final TransportException e) {
+        default void onShardFailedFailure(final Exception e) {
```

Comment:
@bleskes I think we can say what exceptions are *not* bubbled up, but not what exceptions are?